plugins/lookup/onepassword.py

```diff
@@ -495,32 +518,48 @@ def assert_logged_in(self):
         return False
 
     def full_signin(self):
-        required_params = [
-            "subdomain",
-            "username",
-            "secret_key",
-            "master_password",
-        ]
-        self._check_required_params(required_params)
+        if self.service_account_token:
+            environment_update = {"OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token}
+            args = [
+                "whoami",
+            ]
 
-        args = [
-            "account", "add", "--raw",
-            "--address", "{0}.{1}".format(self.subdomain, self.domain),
-            "--email", to_bytes(self.username),
-            "--signin",
-        ]
+            return self._run(args, environment_update=environment_update)
+        else:
+            required_params = [
+                "subdomain",
+                "username",
+                "secret_key",
+                "master_password",
+            ]
+            self._check_required_params(required_params)
 
-        environment_update = {"OP_SECRET_KEY": self.secret_key}
-        return self._run(args, command_input=to_bytes(self.master_password), environment_update=environment_update)
+            args = [
+                "account", "add", "--raw",
+                "--address", "{0}.{1}".format(self.subdomain, self.domain),
+                "--email", to_bytes(self.username),
+                "--signin",
+            ]
+
+            environment_update = {"OP_SECRET_KEY": self.secret_key}
+
+            return self._run(args, command_input=to_bytes(self.master_password), environment_update=environment_update)
 
     def get_raw(self, item_id, vault=None, token=None):
         args = ["item", "get", item_id, "--format", "json"]
         if vault is not None:
             args += ["--vault={0}".format(vault)]
+        if not self.service_account_token and token is not None:
+            args += [to_bytes("--session=") + token]
+        if self.service_account_token:
+            environment_update = {"OP_SERVICE_ACCOUNT_TOKEN": self.service_account_token}
+            return self._run(args, environment_update=environment_update)
+            
         if token is not None:
             args += [to_bytes("--session=") + token]
-
+            
         return self._run(args)
+            return self._run(args)
```

Comment:
GitHub is weird about suggestions on deleted lines. This is a bad merge of my suggestion. Sorry about that.

Reply:
No worries. already removed it.